src/OpenIddict.Core/OpenIddictProvider.Authentication.cs

```diff
@@ -162,13 +177,20 @@ public partial class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServ
             // continue to the next middleware in the pipeline.
             var user = await services.Users.GetUserAsync(principal);
             if (user == null) {
+                logger.LogWarning("Unable to retrieve user from the current principal.");
```

Comment:
Sounds like an error to me. Maybe we should also mention the most common case of this error: when the user was removed from the DB.